src/esp/scene/SceneNode.h

```diff
@@ -88,6 +140,50 @@ class SceneNode : public MagnumObject,
   //! this node is the root
   const Magnum::Range3D& getCumulativeBB() const { return cumulativeBB_; };
 
+  /**
+   * @brief Return SensorSuite containing references to Sensors this SceneNode
+   * holds
+   */
+  esp::sensor::SensorSuite& getNodeSensorSuite() { return *nodeSensorSuite_; }
+
+  /**
+   * @brief Return map containing references to Sensors this SceneNode holds
+   * keys are uuid strings, values are references to Sensors with that uuid
+   */
+  std::map<std::string, std::reference_wrapper<esp::sensor::Sensor>>&
+  getNodeSensors();
+
+  /**
+   * @brief Return SensorSuite containing references to superset of all Sensors
+   * held by this SceneNode and its children
+   */
+  esp::sensor::SensorSuite& getSubtreeSensorSuite() {
+    return *subtreeSensorSuite_;
+  }
+
+  /**
+   * @brief Return map containing references to superset of all Sensors held by
+   * this SceneNode and its children values keys are uuid strings, values are
+   * references to Sensors with that uuid
+   */
+  std::map<std::string, std::reference_wrapper<esp::sensor::Sensor>>&
+  getSubtreeSensors();
+
+  /** @brief Remove sensors in this' nodeSensorSuite from parent's
+   * nodeSensorSuite
```

Comment:
Add a comment, something like:
Note, you only have to call this function when this node is a leaf node.